lib/response.js

```diff
@@ -478,16 +478,19 @@ module.exports = {
 
   /**
    * Checks if the request is writable.
-   * Tests for the existence of the socket
-   * as node sometimes does not set it.
    *
    * @return {Boolean}
    * @api private
    */
 
   get writable() {
+    // can't write any more after response finished
+    if (this.res.finished) return false;
+
     var socket = this.res.socket;
-    if (!socket) return false;
+    // There are already pending outgoing res, but still writable
+    // https://github.com/nodejs/node/blob/v4.4.7/lib/_http_server.js#L486
+    if (!socket) return true;
```

Comment:
wait. not sure if [this](https://github.com/nodejs/node/blob/v4.4.7/lib/_http_server.js#L433) will make res.socket not exists, but this too extreme anyway.